modules/Movecraft/src/main/java/net/countercraft/movecraft/sign/SpeedSign.java

```diff
@@ -40,4 +45,36 @@ public final void onSignTranslate(SignTranslateEvent event) {
         event.setLine(2,String.format("%.2f",craft.getMeanCruiseTime() * 1000) + "ms");
         event.setLine(3,craft.getTickCooldown() + "T");
     }
+
+    @EventHandler
+    public void onSignClick(PlayerInteractEvent event) {
+        if (event.getAction() != Action.RIGHT_CLICK_BLOCK) {
+            return;
+        }
+        Block block = event.getClickedBlock();
+        if (block.getType() != Material.SIGN_POST && block.getType() != Material.WALL_SIGN) {
+            return;
+        }
+        Sign sign = (Sign) event.getClickedBlock().getState();
+        if (!ChatColor.stripColor(sign.getLine(0)).equalsIgnoreCase("Speed:")) {
+            return;
+        }
+        Player player = event.getPlayer();
+        Craft craft = CraftManager.getInstance().getCraftByPlayer(player);
+        if (craft == null) {
+            return;
+        }
+        final int gearShifts = craft.getType().getGearShifts();
+        int currentGear = craft.getCurrentGear();
+        if (gearShifts == 1) {
+            player.sendMessage(I18nSupport.getInternationalisedString("Gearshift - Disabled for craft type"));
+            return;
+        }
+        currentGear++;
+        if (currentGear > gearShifts)
+            currentGear = 1;
+        player.sendMessage(I18nSupport.getInternationalisedString("Gearshift - Gear changed") + " " + currentGear + " / " + gearShifts);
```

Comment:
It would be preferable to send this via the action bar. You can do `player.spigot().sendMessage(ChatMessageType.ACTION_BAR, new TextComponent("text here")));`